### ISSUE TITLE
CSS: Class names and tagname mixed usage, nesting, conventions

### ISSUE THREAD

**Issue by iamzozo** [woocommerce/woocommerce#17228]
I have found some parts in the css/sass files, where structuring and naming classes seems inconsistent a bit.

## Here are some examples and my concerns:

- Mixed usage of underscores and dashes
- Using html tagnames
- Prefixes and without prefixes
- BEM

```
.widget_layered_nav_filters
nav.woocommerce-pagination 
div.product
ul#shipping_method
woocommerce-product-gallery--columns-5
```

However, I agree that prefixing with an html tag speaks itself, how it will be displayed (inline, block, paragraph margins, etc.). But that doesn't make a searchable, overwriteable consistent css, when working with theme development. Understand it's probably about backward compatibility and it's hard to manage these issues. 

## Please take into consideration:
- Remove tagnames from classnames
- Classes with dashes
- Create compact styleguide, how new templates, classes should be named when contributing, and how to arrange them.
- /Refactor current styles in a standardized way/

**TLDR;**

**With the prefixed, block-helper naming convention, with limit class nesting to two levels:**

Original:

```
div.product
    .product_title
    span.price
    p.price
    .woocommerce-product-rating
    div.images
        div.thumbnails
```

Updated:
```
.wc-product
    .product-title
    .product-price
    .product-rating
    .product-images
    .product-thumbnails
```
Let's assume, we have other product classes from another plugin, then prefix and flatten the one above (probably create a separate product.scss)  and encapsulate the woocommerce styling:

```
.wc-product
.wc-product-title
.wc-product-price
.wc-product-rating
.wc-product-images
.wc-product-thumbnails
```

At first sight it seems not too DRY, but the flattened way is more searchable, also readable in the IDE's structure view, and would be easier to work with (building themes) without building the same nesting structure (tagname, #id, .classname):

```
div.product  {
     // ...
     .woocommerce-product-rating {
         color: red;
     }
     // ...
}
```

Could be:
```
.wc-product-rating {
   color: red;
}
```

**Comment by mikejolley** [comment]
👍 Completely agree and we're aware. This is the same as https://github.com/woocommerce/woocommerce/issues/10674 We're not tracking this with a single issue because the effort is ongoing and gradual.

We need to avoid a complete refactor because of the implications to themes + extensions, but we're gradually converting to a standard and will eventually drop old class names in a major update. It's not worth the disruption at present. 

This is our wiki on class naming: https://github.com/woocommerce/woocommerce/wiki/CSS-SASS-coding-guidelines-and-naming-conventions We're using hyphens and BEM.

If you see small changes that can be made to improve things without breaking compat, it would be great to get some PRs in to help with this. But keep changes small :)

Additionally if you see anything from the WIKI feel free to edit or comment here and we'll update it.

Thanks!
